src/main/java/io/xpring/xrpl/model/XRPCurrency.java

```diff
@@ -0,0 +1,34 @@
+package io.xpring.xrpl.model;
+
+import com.google.protobuf.ByteString;
+import org.xrpl.rpc.v1.Currency;
+import org.immutables.value.Value;
+
+/**
+ * An issued currency on the XRP LEdger
+ *
+ * @see "https://xrpl.org/currency-formats.html#currency-codes"
+ */
+@Value.Immutable
+public interface XRPCurrency {
+    static ImmutableXRPCurrency.Builder builder() {
+        return ImmutableXRPCurrency.builder();
+    }
+
+    /**
+     * @return 3 character currency ASCII code
+     */
+    String name();
+
+    /**
+     * @return 160 bit currency code. 20 bytes
+     */
+    ByteString code();
```

Comment:
`ByteString` is a google protobuf thing, right? I don't have any experience with them, so might not be a huge help here.  The other option is to make this a `byte[]`, which seems more native java-y to me.  Could be convinced otherwise.

In your `from` method, you could then just do `.code(currency.getCode().toByteArray())`

Reply:
Love it.  I think making these objects as native-java-y as possible is the goal here, and it doesn't make things much uglier, so I'm going to make that change.